code/modules/vending/_vending.dm

```diff
@@ -819,7 +819,7 @@
 				post_crush_living(living_target, was_alive)
 				flags_to_return |= (SUCCESSFULLY_CRUSHED_MOB|SUCCESSFULLY_CRUSHED_ATOM)
 
-			else if (atom_target.uses_integrity && !(atom_target.invisibility > SEE_INVISIBLE_LIVING) && !(is_type_in_typecache(atom_target, GLOB.WALLITEMS_INTERIOR) || is_type_in_typecache(atom_target, GLOB.WALLITEMS_EXTERIOR)))
+			else if (atom_target.uses_integrity && !(atom_target.invisibility > SEE_INVISIBLE_LIVING) && !(istype(atom_target, /obj/structure/chair)) && !(istype(atom_target, /obj/machinery/conveyor)) && !(is_type_in_typecache(atom_target, GLOB.WALLITEMS_INTERIOR) || is_type_in_typecache(atom_target, GLOB.WALLITEMS_EXTERIOR)))
```

Comment:
1. This check is getting very ugly, perhaps separate it out into its own proc `/proc/can_squish_object()`?
2. Should probably have a typecache for items not to cruch, rather than doing 2 istypes, because I feel like this will be expanded in the future

Reply:
Fair, I can do that

Reply:
I hope a normal list is ok instead of a typecash

Reply:
a typecache is really easy to make just look up instances of `typecacheof` in the code.

also you still should spin this out into its own proc

Reply:
how do I split it into a separate proc